saleor/product/views.py

```diff
@@ -120,18 +121,20 @@ def category_index(request, path, category_id):
                         category_id=category_id)
     products = (products_with_details(user=request.user)
                 .filter(categories__name=category)
-                .order_by(DEFAULT_SORT))
+                .order_by('name'))
     product_filter = ProductFilter(
         request.GET, queryset=products, category=category)
     products_paginated = get_paginator_items(
         product_filter.qs, PAGINATE_BY, request.GET.get('page'))
     products_and_availability = list(products_with_availability(
         products_paginated, request.discounts, request.currency))
-    sort_by = request.GET.get('sort_by', DEFAULT_SORT)
+    sort_by = request.GET.get(
+        'sort_by', pgettext_lazy('Product field', 'name'))
```

Comment:
Never access `request.(...)` directly in view code. Use a form and validate the value. Even if it's a single line. Currently I can pass arbitrary values as `sort_by` and it will work.

Reply:
Also why do we pass _translated_ strings in URLs?

Reply:
Ok, i will create new form just for checking sort_by in GET.

Reply:
We do not, they are not translated.

Reply:
So `sort_by` sometimes contains an _untranslated_ string from the browser and sometimes it contains a _translated_ default value?

Reply:
Imagine that `sort_by` choices are the following:

```python
SORTBY_CHOICES = [
    ('name', pgettext_lazy('Product list sorting options', 'Name'),
    ('price', pgettext_lazy('Product list sorting options', 'Price')]
```

Which value would be passed as `sort_by=...` in URL?

Reply:
No the flow was to translate value of GET 'sort_by' field. But as you mentioned it was dynamical.